napari/_qt/layer_controls/qt_tracks_controls.py

```diff
@@ -133,6 +144,19 @@ def _on_tail_length_change(self, event=None):
             value = np.clip(value, 1, MAX_TAIL_LENGTH)
             self.tail_length_slider.setValue(value)
 
+    def _on_head_length_change(self, event=None):
+        """Receive layer model track line width change event and update slider.
+
+        Parameters
+        ----------
+        event : qtpy.QtCore.QEvent, optional.
+            Event from the Qt context, by default None.
+        """
+        with self.layer.events.head_length.blocker():
+            value = self.layer.head_length
+            value = np.clip(value, 0, MAX_TAIL_LENGTH)
```

Comment:
It feels a bit odd to me that the clipping is done in the layer controls. This means that the controls and layer model can become out of sync (e.g., set the `head_length` value to greater than `MAX_TAIL_LENGTH` on the layer.

Reply:
I now see that this is the case for the `tail_length` already. I think the clipping should be done on the layer so that the layer and controls stay in sync. Thoughts? cc @sofroniewn

Reply:
yeah - we've run into problems with this in other parts of the codebase with sliders and their max length - i think the paintbrush size slider was the last place. I can't remember right now what we did there though, does @ppwadhwa @DragaDoncila or @tlambert03 recall - i think some of you were involved in that work

Reply:
yeah, it's a bit of a bummer that our layers aren't pydantic models, ideally the declaration of `head_length` on the model would be a `conint` that contains this information, and widgets could then easily represent it (with a single source of truth).  Until then, I don't have an elegant solution... in other places, we've added annoying logic to allow the widget to "grow" it's range to include whatever value has been provided programatically.

In any case, I agree with @kevinyamauchi that keeping the model and the view in sync is the main priority.  If you clip it on the widget, you have to then clip it on the model too

Reply:
yeah, we can do a clip on model, though we might want to emit a warning